app/Http/Transformers/PostTransformer.php

```diff
@@ -48,11 +47,16 @@ public function transform(Post $post)
                 'total' => $post->reactions_count,
             ],
             'status' => $post->status,
-            'location' => $post->location,
             'created_at' => $post->created_at->toIso8601String(),
             'updated_at' => $post->updated_at->toIso8601String(),
+            'location' => $post->location,
         ];
 
+        // If the post isn't anonymous or if this is the owner of the post, return PII information.
+        if ((! $post->actionModel->anonymous && ! is_null($post->actionModel->anonymous)) || Gate::allows('viewAll', $post) {
```

Comment:
I think this conditional can still be simplified a bit:

```php
if ((! $post->actionModel->anonymous || Gate::allows('viewAll', $post)) {
```

Reply:
@DFurnes the reason I put `! is_null($post->actionModel->anonymous)` is because I'm not sure if after we run this, existing records will have `NULL` for the `anonymous` column.  If they are `NULL`, I think we'd want to check this?

Reply:
That should be okay, because we want the default `null` to work the same way as if `anonymous` was set to false. In both cases (`! false` and `! null`) this evaluates to `true`.

Reply:
oh true! Updated :)